emitter.py

```diff
@@ -129,7 +120,60 @@ def http_emitter(message, log, agentConfig, endpoint):
         raise Exception("Posting payload failed")
 
 
-def post_headers(agentConfig, payload):
+def split_payload(legacy_payload):
+    metrics_payload = {"series": []}
+
+    # See https://github.com/DataDog/dd-agent/blob/5.11.1/checks/__init__.py#L905-L926 for format
+    for ts in legacy_payload['metrics']:
+        sample = {
+            "metric": ts[0],
+            "points": [(ts[1], ts[2])],
+            "source_type_name": "System",
+        }
+
+        if len(ts) >= 4:
+            # Default to the metric hostname if present
+            if ts[3].get('hostname'):
+                sample['host'] = ts[3]['hostname']
+            else:
+                # If not use the general payload one
+                sample['host'] = legacy_payload['internalHostname']
+
+            if ts[3].get('type'):
+                sample['type'] = ts[3]['type']
+            if ts[3].get('tags'):
+                sample['tags'] = ts[3]['tags']
+            if ts[3].get('device_name'):
+                sample['device'] = ts[3]['device_name']
+
+        metrics_payload["series"].append(sample)
+
+    del legacy_payload['metrics']
+
+    return legacy_payload, metrics_payload
+
+def http_emitter(message, log, agentConfig, endpoint):
+    api_key = message.get('apiKey')
+
+    if not api_key:
+        raise Exception("The http emitter requires an api key")
+
+    # For perf reason. We now want to send the metrics to the api endpoint. So we are extracting them
+    # from the payload here, transform them into the expected format and send them (via the forwarder)
+
+    legacy_url = "{0}/intake/{1}?api_key={2}".format(agentConfig['dd_url'], endpoint, api_key)
+    metrics_endpoint = "{0}/api/v1/series?api_key={1}".format(agentConfig['dd_url'], api_key)
+
+    legacy_payload, metrics_payload = split_payload(message)
+
+    # Post legacy payload
+    post_payload(legacy_url, legacy_payload, agentConfig, log)
+
+    # Post metrics payload
+    post_payload(metrics_endpoint, metrics_payload, agentConfig, log)
```

Comment:
we probably have to check whether `simulator record -t <file>` still works

Reply:
I'm not expecting any problems, I guess I'm just putting stickies here ;)

Reply:
done that